pvlib/modelchain.py

```diff
@@ -1118,9 +1118,7 @@ def infer_temperature_model(self):
             array.temperature_model_parameters for array in self.system.arrays)
         params = _common_keys(temperature_model_parameters)
         # remove or statement in v0.9
```

Comment:
Let's delete this comment as well, since we are now removing the specified `or`

Reply:
Will do, nice catch about removing the comment about the `or` statement.

Thanks for creating the test.  I have one comment about it before implementing.  Since the `PVSystem` [docstring ](https://github.com/pvlib/pvlib-python/blob/main/pvlib/pvsystem.py#L109-L121) states:
```
If `arrays` is
        specified the following PVSystem parameters are ignored:
```

I think we shouldn't even include the `temperature_model_parameters` argument in the test to make it more clear in case future developers are examining it.  So the following change:
```
system = pvsystem.PVSystem(arrays,
                               inverter_parameters={'pdc0': 1})
```

Reply:
It's true that passing `temperature_model_parameters` alongside `arrays` is not a valid use for PVSystem.  If a user did this, they should not expect it to work.  But the intent of this test is to make sure that, when this incorrect usage occurs, the correct error is raised: `ValueError: could not infer temperature...` instead of `AttributeError: 'PVSystem' object has no attribute 'racking_model'`.  What you suggest (do not specify `temperature_model_parameters` at all) would be a verification of some other behavior, not that the issue from #1759 is resolved.

If you like, please clarify the explanation comment in the test for future readers :)

Reply:
I think I was updating my comment while you were replying.  I'm not proposing modifying the test by not specifying `temperature_model_parameters`.  I'm proposing updated the error message to make it more informative, as well as fixing the last sentence.

Reply:
Updating that error message is the issue in #1946, right?  I think we should fix one thing at a time, and this PR should focus just on #1759.

Reply:
For [this ValueError](https://github.com/pvlib/pvlib-python/blob/b67a668378ecb66dfd5d2be6273e22827776a858/pvlib/modelchain.py#L1134), what about

"Could not infer the temperature model. If Arrays are not used to construct the PVsystem, check the PVsystem attributes racking_model and module_type. If Arrays are used, check that all Arrays in ModelChain.system.arrays have parameters for the same temperature model.  Common temperature model parameters: {params}."

The `params` dict is what ModelChain found when it tries to infer the temperature model, and is provided here to (hopefully) help the user to diagnose and fix the error.

Reply:
@kandersolar, you are correct, makes sense, thanks.  Will incorporate as per your suggestion and then after we finish probably tackle #1946  to address the error message.  Thanks @cwhanse for the phrasing.